chrome/content/zotero/xpcom/utilities_internal.js

```diff
@@ -94,13 +94,17 @@ Zotero.Utilities.Internal = {
 	 * @return {Promise<Object|null>} Item metadata in translator format,
 	 * or null if nothing was translated
 	 */
-	'translateURL': async function (url) {
-		let doc = (await Zotero.HTTP.processDocuments(url, doc => doc))[0];
-		let newItems = await this.translateDocument(doc);
-		if (!newItems.length) {
+	translateURL: async function (url) {
+		try {
```

Comment:
URL translation errors should be handled because it's expected that some publisher websites will be down therefore this shouldn't prevent to use other metadata update methods. While our identifier resolver API and other identifier APIs should fail hard and stop metadata updating for the current item.

Reply:
I don't think the `try` statements belong in these functions. Calling code should be able to know that an attempt failed and handle it however it wants, whether that's retrying or moving on to another method.